public/app/plugins/panel/state-timeline/timeline.ts

```diff
@@ -319,9 +319,10 @@ export function getConfig(opts: TimelineCoreOptions) {
               for (let ix = 0; ix < dataY.length; ix++) {
                 if (dataY[ix] != null) {
                   const boxRect = boxRectsBySeries[sidx - 1][ix];
+                  const txt = formatValue(sidx, dataY[ix]);
+                  const valueRect = u.ctx.measureText(txt);
```

Comment:
Thanks for the change here. It's definitely not a bad change and we've made similar changes elsewhere for label measurements. There could be a performance concern here as `context.measureText()` is fairly slow and on panels with a large number of data points this could cause issues.

@leeoniya do you think this would be too much of a hit in this case?

Reply:
> @leeoniya do you think this would be too much of a hit in this case?

yeah, i think we'll need to wrap this in some imperfect character length check precheck.